.github/workflows/zfs-linux-tests.yml

```diff
@@ -0,0 +1,113 @@
+name: zfs-linux-tests
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        description: 'The ubuntu version: 20.02 or 22.04'
+        required: true
+        type: string
+
+jobs:
+
+  zloop:
+    runs-on: ubuntu-${{ inputs.os }}
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: modules-${{ inputs.os }}
+    - name: Install modules
+      run: |
+        tar xzf modules-${{ inputs.os }}.tgz
+        .github/workflows/scripts/setup-dependencies.sh tests
+    - name: Tests
+      timeout-minutes: 30
+      run: |
+        sudo mkdir -p /var/tmp/zloop
+        # run for 10 minutes or at most 2 iterations for a maximum runner
+        #   time of 20 minutes.
+        sudo /usr/share/zfs/zloop.sh -t 600 -I 2 -l -m1 -- -T 120 -P 60
+    - name: Prepare artifacts
+      if: failure()
+      run: |
+        sudo chmod +r -R /var/tmp/zloop/
+    - uses: actions/upload-artifact@v3
+      if: failure()
+      with:
+        name: Zpool-logs-${{ inputs.os }}
+        path: |
+          /var/tmp/zloop/*/
+          !/var/tmp/zloop/*/vdev/
+        if-no-files-found: ignore
+    - uses: actions/upload-artifact@v3
+      if: failure()
+      with:
+        name: Zpool-files-${{ inputs.os }}
+        path: |
+          /var/tmp/zloop/*/vdev/
+        if-no-files-found: ignore
+
+  sanity:
```

Comment:
IIRC we discussed that now we may not even need sanity tests, but this PR is already to good to be true (thanks @mcmilk ) so let it be